mp2-v1/src/api.rs

```diff
@@ -201,15 +201,21 @@ pub enum SlotInputs {
 
 /// Compute metadata hash for a "merge" table. Right now it supports only merging tables from the
 /// same address.
-pub fn merge_metadata_hash(
+pub fn merge_metadata_hash<const MAX_COLUMNS: usize, const MAX_FIELD_PER_EVM: usize>(
     contract: Address,
     chain_id: u64,
     extra: Vec<u8>,
     table_a: SlotInputs,
     table_b: SlotInputs,
 ) -> MetadataHash {
-    let md_a = value_metadata(table_a, &contract, chain_id, extra.clone());
-    let md_b = value_metadata(table_b, &contract, chain_id, extra);
+    let md_a = value_metadata::<MAX_COLUMNS, MAX_FIELD_PER_EVM>(
+        table_a,
+        &contract,
+        chain_id,
+        extra.clone(),
+    );
+    let md_b =
+        value_metadata::<MAX_COLUMNS, MAX_FIELD_PER_EVM>(table_b, &contract, chain_id, extra);
     let combined = md_a + md_b;
```

Comment:
@nikkolasg do you think it is still safe to simply aggregate metadata digests in merge table case? I was wondering if we should compute this as `D(md_a) + D(md_b)` to avoid that the metadata digest of columns related to the mapping gets mixed with metadata digest of columns related to single variables, but I don't find any way to build the tables in a different way even if we keep the combined digest simply as `md_a + md_b`. But I would like a double check on this. We could also consider to compute the combined digest anyhow as `D(md_a) + D(md_b)` to be safer than sorry, though we are not sure it's necessary.

Reply:
So @silathdiir, as agreed offline, it would be better to compute the combined digest as `D(md_a) + D(md_b)`. This would need to be changed also in the merge circuit